datastore/tests/unit/test_helpers.py

```diff
@@ -135,6 +135,26 @@ def test_mismatched_value_indexed(self):
         with self.assertRaises(ValueError):
             self._call_fut(entity_pb)
 
+    def test_index_mismatch_ignores_empty_list(self):
+        from google.cloud.datastore_v1.proto import entity_pb2
+        from google.cloud.datastore.helpers import _new_value_pb
+
+        _PROJECT = 'PROJECT'
+        _KIND = 'KIND'
+        _ID = 1234
+        entity_pb = entity_pb2.Entity()
+        entity_pb.key.partition_id.project_id = _PROJECT
+        entity_pb.key.path.add(kind=_KIND, id=_ID)
+
+        array_val_pb = _new_value_pb(entity_pb, 'baz')
+        array_pb = array_val_pb.array_value.values
+        array_val_pb.array_value.CopyFrom(entity_pb2.ArrayValue(values=[]))
+        
+        unindexed_value_pb1 = array_pb.add()
```

Comment:
We don't want to assign any value into the `array_pb`:  it won't be empty if we do.

I'm not sure whether this line is doing nothing (because 151 might replace the `array_val_pb.array_value.values`) or whether it is actually making a non-empty array.

Reply:
I think you are right.  I had a previous test that I was using, and I think L150 and L153 are redundant and I have removed them.

Reply:
Actually, might as well make a note here, the array_pb.add() is how I got this I mentioned above, which cannot be used to create an entity at this time.  I changed the code in the original post to fix it, but I figured if it isn't broken, don't fix it.  But this can be a note in case we need to in the future.
```
array_values{
  values{
  }
}
```